### PR TITLE
Choice rule migration

### PR DESCRIPTION
**Pull request by dhuebner**
Choice rules like described in #40

### CODE REVIEW

File: com.regnosys.rosetta.lib/src/main/java/model/basictypes.rosetta [hunk 1]
```diff
@@ -6,6 +6,7 @@ basicType int
 basicType number
 basicType boolean
 basicType time
+basicType any
```
Comment (jim-h-wang) [inline com.regnosys.rosetta.lib/src/main/java/model/basictypes.rosetta]: I think we removed the need for the `any` type, by keeping one-of and choice rules as language features. So we should either remove the basic type here or have validation to stop usage outside of the core library.

File: com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext [hunk 2]
```diff
@@ -74,7 +74,22 @@ ShortcutDeclaration:
 Condition:
 	'condition' RosettaNamed? ':' Definable?
 		Annotations?
-		(expressions+=RosettaCalcExpression ';')+
+		( constraint = OneOf
```
Comment (jim-h-wang) [inline com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext]: Can we add a validation warning if multiple `one-of`s are used on a `data` object

File: com.regnosys.rosetta/src/com/regnosys/rosetta/RosettaExtensions.xtend [hunk 3]
```diff
@@ -221,4 +230,21 @@ class RosettaExtensions {
 		}
 		return false
 	}
+	
+	def boolean isChoiceRuleCondition(Condition cond) {
+		return cond.constraint !== null
```
Comment (jim-h-wang) [inline com.regnosys.rosetta/src/com/regnosys/rosetta/RosettaExtensions.xtend]: Something doesn't feel right about this method. We are either testing for existence of Constraints, in which case the name of the function should reflect that; or we are checking for choice rules, in which case the predicate needs to update?

File: com.regnosys.rosetta.lib/src/main/java/model/basictypes.rosetta (hunk 1 above)
Comment (dhuebner) [reply com.regnosys.rosetta.lib/src/main/java/model/basictypes.rosetta]: You are right! Just forgot to remove. I will replace it with `empty` later

File: com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext (hunk 2 above)
Comment (dhuebner) [reply com.regnosys.rosetta/src/com/regnosys/rosetta/Rosetta.xtext]: SureI will create a new issue to collect validation rules

File: com.regnosys.rosetta/src/com/regnosys/rosetta/RosettaExtensions.xtend (hunk 3 above)
Comment (dhuebner) [reply com.regnosys.rosetta/src/com/regnosys/rosetta/RosettaExtensions.xtend]: It describes the difference between Data rules and Choice rules. That is something you need to consolidate as one validation-concept; over all layers.

Comment (jim-h-wang) [reply com.regnosys.rosetta/src/com/regnosys/rosetta/RosettaExtensions.xtend]: But `cond.constraint !== null` includes Choice rules as well as OneOfs, so we are treating those 2 things as the same thing in this context. Let's take the naming conversation offline and we can address in subsequence PR.

**Pull request merged**